tools/PI/DevHome.PI/SettingsUi/AddToolControl.xaml.cs

```diff
@@ -0,0 +1,119 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.Runtime.InteropServices;
+using DevHome.PI.Helpers;
+using Microsoft.UI.Xaml;
+using Microsoft.UI.Xaml.Controls;
+using Windows.Win32;
+using Windows.Win32.UI.Controls.Dialogs;
+
+namespace DevHome.PI.SettingsUi;
+
+public sealed partial class AddToolControl : UserControl
+{
+    public AddToolControl()
+    {
+        InitializeComponent();
+    }
+
+    private void BrowseButton_Click(object sender, RoutedEventArgs e)
+    {
+        HandleBrowseButton();
+    }
+
+    private void HandleBrowseButton()
+    {
+        // Unfortunately WinUI3's OpenFileDialog does not work if we're running elevated. So we have to use the old Win32 API.

```

Comment:
consider using more nuetral language here, e.g. exlucing Unfortunately, or "As it turns out"